sovryn-token-bridge/federator/src/lib/Federator.js

```diff
@@ -157,55 +157,72 @@ module.exports = class Federator {
         this.logger.info('get transaction id:', transactionId);
 
         let wasProcessed = await this.federationContract.methods.transactionWasProcessed(transactionId).call();
+        this.logger.info('wasProcessed:', wasProcessed);
         if (!wasProcessed) {
-            let hasVoted = await this.federationContract.methods.hasVoted(transactionId).call({from: from});
-            if (!hasVoted) {
-                this.logger.info(`Voting tx: ${log.transactionHash} block: ${log.blockHash} token: ${symbol}`);
-                await this._voteTransaction(tokenAddress,
-                    receiver,
-                    amount,
-                    symbol,
-                    log.blockHash,
-                    log.transactionHash,
-                    log.logIndex,
-                    decimals,
-                    granularity,
-                    userData
-                );
-            } else {
-                this.logger.debug(`Block: ${log.blockHash} Tx: ${log.transactionHash} token: ${symbol}  has already been voted by us`);
+            this.logger.info('getting transactionIdU:', {
+                tokenAddress,
+                receiver,
+                amount,
+                symbol,
+                blockHash: log.blockHash,
+                transactionHash: log.transactionHash,
+                logIndex: log.logIndex,
+                decimals,
+                granularity,
+                userData
+            });
+
+            let transactionIdU = await this.federationContract.methods.getTransactionIdU(
+                tokenAddress,
+                receiver,
+                amount,
+                symbol,
+                log.blockHash,
+                log.transactionHash,
+                log.logIndex,
+                decimals,
+                granularity,
+                userData || []
+            ).call();
+
+            this.logger.info('get transaction id U:', transactionIdU);
+            let wasProcessedU = await this.federationContract.methods.transactionWasProcessed(transactionIdU).call();
+            this.logger.info('wasProcessedU:', wasProcessedU);
```

Comment:
@bravepooh and here you see if the transaction was processed after checking by transactionIdU (Boolean)